src/glide/browser/base/content/test/hints/browser_hints.ts

```diff
@@ -578,3 +578,38 @@ add_task(async function test_hint_action_function__bad_return() {
     );
   });
 });
+
+add_task(async function test_clear_no_hints_notification_on_retrigger() {
+  await BrowserTestUtils.withNewTab(FILE, async (_) => {
+    GlideBrowser.api.hints.show({ selector: "[data-no-such-element]" });
+
+    await TestUtils.waitForCondition(
+      () => gNotificationBox.getNotificationWithValue("glide-no-hints-found") !== null,
+      "Waiting for 'No hints found' notification to appear",
+    );
+
+    Assert.notStrictEqual(
+      gNotificationBox.getNotificationWithValue("glide-no-hints-found"),
+      null,
+      "Notification should be present after no hints found",
+    );
```

Comment:
nit: instead of duplicating the checks like this (you might've taken inspiration from some outdated code, sorry!) there is a `waiter()` API you can use instead e.g.

```suggestion
    await waiter(() => gNotificationBox.getNotificationWithValue("glide-no-hints-found")).ok(
      "Waiting for 'No hints found' notification to appear",
    );
```

This applies to the below assertion as well.